### DISCUSSION TITLE
additional field validation

### DISCUSSION THREAD

**Discussion by Halvanhelv** [janko/rodauth-rails#209] [General]
Hi! I'm still learning rodauth and I like it a lot, I wanted to know how to do validation correctly

1) how can i validate additional fields when registering, for example i need to check if the nickname field is not obscene, at the moment i can do it only in a very strange way by calling validate on the model in the hook `before_create_account`, i dont like this solution because since I can't check only part of the fields, I also don't want to write a separate controller for registering the account, maybe someone has seen a similar case and knows a better solution?

2) how to insert errors in bulk if some fields are invalid at the same time, at the moment I did so 

```
    before_create_account do
      set_field_error(:first_name, 'First name cannot be empty') if param_or_nil('first_name').blank?
      set_field_error(:last_name, 'Last name cannot be empty') if param_or_nil('last_name').blank?
      if @field_errors.any?
        throw_error_status(422, '', '')
      end
    end
```

maybe the author or contributors know a better way?

Thanks for help

**Comment by janko** [comment]
> at the moment i can do it only in a very strange way by calling validate on the model in the hook before_create_account, i dont like this solution because since I can't check only part of the fields

I didn't quite understand why you don't prefer this solution. What do you mean by "part of the fields"?

> how to insert errors in bulk if some fields are invalid at the same time

Rodauth doesn't currently have a built-in way to add multiple validation errors, because it itself always returns a single validation error. I think the reason is that validations often have dependencies – e.g. you can't validate an account's password if the email wasn't found – and it's simpler to just always throw. If you want Rodauth to support this, I recommend asking on that repo.

**Reply by Halvanhelv** [reply]
> I didn't quite understand why you don't prefer this solution. What do you mean by "part of the fields"?

I think this part of the question is not relevant anymore and can be reduced to - how to validate data if I need to check the parameters assigned, for example whether the nickname of a user contains insults

that is to say what is the best way to call validation
